File: src/main/java/com/example/lms/service/LeaveRequestService.java
package com.example.lms.service;

import com.example.lms.entity.LeaveRequest;

import java.util.List;

public interface LeaveRequestService {
    List<LeaveRequest> getAllLeaveRequests();

    LeaveRequest getLeaveRequestById(int leaveRequestId);

    LeaveRequest saveLeaveRequest(LeaveRequest leaveRequest);

    LeaveRequest updateLeaveRequest(int requestId, LeaveRequest leaveRequest);

    void deleteLeaveRequest(int leaveRequestId);

    List<LeaveRequest> getLeaveRequestsByEmployeeId(int employeeId);
    LeaveRequest updateLeaveRequestStatus(int requestId, String newStatus);

}

File: leavemanagement1ui/src/services/LeaveRequestService.js
import axios from 'axios';

const LEAVEREQUEST_API_BASE_URL = "http://localhost:8089/api/leave-requests";

class LeaveRequestService {
  saveLeaveRequest(leaveRequest) {
    return axios.post(LEAVEREQUEST_API_BASE_URL, leaveRequest);
  }

  getAllLeaveRequests() {
    return axios.get(LEAVEREQUEST_API_BASE_URL);
  }

  getLeaveRequestById(leaveRequestId) {
    return axios.get(`${LEAVEREQUEST_API_BASE_URL}/${leaveRequestId}`);
  }

  updateLeaveRequest(leaveRequestId, leaveRequest) {
    return axios.put(`${LEAVEREQUEST_API_BASE_URL}/${leaveRequestId}`, leaveRequest);
  }

  deleteLeaveRequest(leaveRequestId) {
    return axios.delete(`${LEAVEREQUEST_API_BASE_URL}/${leaveRequestId}`);
  }

  getLeaveRequestsByEmployeeId(employeeId) {
    return axios.get(`${LEAVEREQUEST_API_BASE_URL}/employee/${employeeId}`);
  }

  updateLeaveRequestStatus(requestId, statusMap) {
    return axios.put(`${LEAVEREQUEST_API_BASE_URL}/${requestId}/status`, statusMap);
  }
}

export default new LeaveRequestService();
